include/maliput/geometry_base/kd_tree_strategy.h

```diff
@@ -0,0 +1,87 @@
+// BSD 3-Clause License
+//
+// Copyright (c) 2022, Woven Planet.
+// All rights reserved.
+//
+// Redistribution and use in source and binary forms, with or without
+// modification, are permitted provided that the following conditions are met:
+//
+// * Redistributions of source code must retain the above copyright notice, this
+//   list of conditions and the following disclaimer.
+//
+// * Redistributions in binary form must reproduce the above copyright notice,
+//   this list of conditions and the following disclaimer in the documentation
+//   and/or other materials provided with the distribution.
+//
+// * Neither the name of the copyright holder nor the names of its
+//   contributors may be used to endorse or promote products derived from
+//   this software without specific prior written permission.
+//
+// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+// AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+// IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+// DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE
+// FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+// DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+// SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+// CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+// OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+// OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+#pragma once
+
+#include <memory>
+#include <optional>
+#include <vector>
+
+#include "maliput/geometry_base/strategy_base.h"
+#include "maliput/math/kd_tree.h"
+
+namespace maliput {
+namespace geometry_base {
+
+class MaliputPoint : public maliput::math::Vector3 {
+ public:
+  MALIPUT_DEFAULT_COPY_AND_MOVE_AND_ASSIGN(MaliputPoint)
+  MaliputPoint() = default;
```

Comment:
My point here is that we should care about state of objects in their entire lifecycle and make sure operations are leave the object in a consistent state. If not, we could face unexpected problems.

That said, I don't think a `MaliputPoint` should be default constructed. IMO, we need to remove the default constructor and make sure we always pass a point and a lane ID. Otherwise, becasue it is necessary to have a default constructor, let's make sure that the lane ID has a no-value / unintialized state and provide setters to initialize the object.

What do you think?

Reply:
I think that removing `MaliputPoint` default constructor makes a lot of sense taking into account the lifecycle of the instances of that class right now. The points shouldn't be modified and it does not make sense to have a point without any x,y,z or lane_id value. In addition to that, `MaliputPoint` is within the private section of the `KDTreeStrategy` class because it is only used by the KDTree implementation.

Reply:
Done.